### ISSUE TITLE
Issue with Hive + elastic search integration CDH 5.7.1

### ISSUE THREAD

**Issue by anandk89** [elastic/elasticsearch-hadoop#832]
Hello,

I am using below version of stack

1) Hive 1.1.0-cdh5.7.1
2) Elastic search 1.7.3

Jars used
elasticsearch-hadoop-hive-2.3.2.jar

I am facing lots of issues while installing. I got two issue one it is the SAX parser exception for some unknown characters and second as show below.

Caused by: java.lang.ClassNotFoundException: org.elasticsearch.hadoop.hive.EsHiveInputFormat
        at java.net.URLClassLoader$1.run(URLClassLoader.java:366)
        at java.net.URLClassLoader$1.run(URLClassLoader.java:355)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.net.URLClassLoader.findClass(URLClassLoader.java:354)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:425)
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:358)
        at java.lang.Class.forName0(Native Method)
        at java.lang.Class.forName(Class.java:270)
        at org.apache.hive.com.esotericsoftware.kryo.util.DefaultClassResolver.readName(DefaultClassResolver.java:136)
        ... 49 more

I have added the above mentioned jar in $HIVE_HOME/lib and restarted the Hive I am trying to invoke command like  hive -e 'INSERT OVERWRITE TABLE test.ECM_METADATA_SERDE_ES SELECT \* from test.ECM_METADATA_SERDE'. The same thing works well on CDH 4.7 with Hive 1.0 version. Please guide what I am missing.

### What kind an issue is this?
- [x] Bug report. If you’ve found a bug, please provide a code snippet or test to reproduce it below.  
  The easier it is to track down the bug, the faster it is solved. 
- [ ] Feature Request. Start by telling us what problem you’re trying to solve.  
  Often a solution already exists! Don’t send pull requests to implement new features without 
  first getting our support. Sometimes we leave features out on purpose to keep the project small.

### Issue description

Description
### Steps to reproduce

Code:

```
Test/code snippet
```

Strack trace:

```
Stack trace goes here
```
### Version Info

OS:         :  
JVM         :  
Hadoop/Spark:  
ES-Hadoop   :  
ES          :

**Comment by anandk89** [comment]
After adding other jars in dist I am now getting below error.


016-08-25 08:28:09,629 INFO [main] org.apache.hadoop.mapreduce.v2.app.MRAppMaster: Created MRAppMaster for application appattempt_1472110818465_0003_000001
2016-08-25 08:28:10,403 FATAL [main] org.apache.hadoop.conf.Configuration: error parsing conf job.xml
org.xml.sax.SAXParseException; systemId: file:///mnt/resource/yarn/nm/usercache/root/appcache/application_1472110818465_0003/container_1472110818465_0003_01_000001/job.xml; lineNumber: 885; columnNumber: 51; Character reference "&#0" is an invalid XML character.
	at org.apache.xerces.parsers.DOMParser.parse(Unknown Source)
	at org.apache.xerces.jaxp.DocumentBuilderImpl.parse(Unknown Source)
	at javax.xml.parsers.DocumentBuilder.parse(DocumentBuilder.java:150)
	at org.apache.hadoop.conf.Configuration.parse(Configuration.java:2491)
	at org.apache.hadoop.conf.Configuration.loadResource(Configuration.java:2560)
	at org.apache.hadoop.conf.Configuration.loadResources(Configuration.java:2513)
	at org.apache.hadoop.conf.Configuration.getProps(Configuration.java:2409)
	at org.apache.hadoop.conf.Configuration.get(Configuration.java:1233)
	at org.apache.hadoop.mapreduce.v2.util.MRWebAppUtil.initialize(MRWebAppUtil.java:51)
	at org.apache.hadoop.mapreduce.v2.app.MRAppMaster.main(MRAppMaster.java:1465)
2016-08-25 08:28:10,405 FATAL [main] org.apache.hadoop.mapreduce.v2.app.MRAppMaster: Error starting MRAppMaster
java.lang.RuntimeException: org.xml.sax.SAXParseException; systemId: file:///mnt/resource/yarn/nm/usercache/root/appcache/application_1472110818465_0003/container_1472110818465_0003_01_000001/job.xml; lineNumber: 885; columnNumber: 51; Character reference "&#0" is an invalid XML character.
	at org.apache.hadoop.conf.Configuration.loadResource(Configuration.java:2656)
	at org.apache.hadoop.conf.Configuration.loadResources(Configuration.java:2513)
	at org.apache.hadoop.conf.Configuration.getProps(Configuration.java:2409)
	at org.apache.hadoop.conf.Configuration.get(Configuration.java:1233)
	at org.apache.hadoop.mapreduce.v2.util.MRWebAppUtil.initialize(MRWebAppUtil.java:51)
	at org.apache.hadoop.mapreduce.v2.app.MRAppMaster.main(MRAppMaster.java:1465)
Caused by: org.xml.sax.SAXParseException; systemId: file:///mnt/resource/yarn/nm/usercache/root/appcache/application_1472110818465_0003/container_1472110818465_0003_01_000001/job.xml; lineNumber: 885; columnNumber: 51; Character reference "&#0" is an invalid XML character.
	at org.apache.xerces.parsers.DOMParser.parse(Unknown Source)
	at org.apache.xerces.jaxp.DocumentBuilderImpl.parse(Unknown Source)
	at javax.xml.parsers.DocumentBuilder.parse(DocumentBuilder.java:150)
	at org.apache.hadoop.conf.Configuration.parse(Configuration.java:2491)
	at org.apache.hadoop.conf.Configuration.loadResource(Configuration.java:2560)
	... 5 more
2016-08-25 08:28:10,413 INFO [main] org.apache.hadoop.util.ExitUtil: Exiting with status 1


This I found on my couple of nodes log. Kindly Help

**Comment by anandk89** [comment]
@costin : Please help.

**Comment by jbaiera** [comment]
The first issue should be fixed if you've installed the es hadoop jar into the environment following [these instructions](https://www.elastic.co/guide/en/elasticsearch/hadoop/current/hive.html#_installation_3). The second error doesn't seem to be ES-Hadoop related?

    2016-08-25 08:28:10,403 FATAL [main] org.apache.hadoop.conf.Configuration: error parsing conf job.xml

Seems that your job has a malformed Configuration XML. "&#0;" is the NULL character, and is not allowed in XML. Are you able to determine where the character shows up?

**Comment by anandk89** [comment]
Thanks a lot for response @jbaiera 

I noticed that there were some issues while resolving internal IP addresses so I added below property and the cluster is now up and running.

**network.host: _site_**

Thanks again. Only thing bothering is the elasticsearch-hadoop-hive-2.3.2.jar which I added in $HIVE_HOME/lib is not taking effect when I run hive queries using option **hive - e** . So I wrote a script and exported those jars externally in Hive to add data into ES via HIVE. It'll be great help if you provide me pointers on how these files should take effect for mentioned CDH version.
